### DISCUSSION TITLE
Posso retirar o GetById do ITodoRepository e incluir no TodoQueries ?

### DISCUSSION THREAD

**Discussion by rafaelnardomello** [balta-io/7196#5] [Perguntas e Respostas] [answered]
Boa tarde !

Fiquei com uma dúvida em **retirar o GetById do ITodoRepository** e incluir no TodoQueries, pois apesar do GetById estar fazendo uma hidratação e tal, trata-se de uma query. Assim sendo, como estamos segregando as operações de consulta em Queries e as de  escrita, atualização, deleção, em commands, **eu poderia retirar o método GetById do contrato do repositório ITodoRepository e incluí-lo na classe TodoQueries** ? Isso está correto ?

Abs !

//Chamando o método **GetById do _repositório_**, no Handler

 public ICommandResult Handle(MarkToDoAsDoneCommand command)
       {
            //Fail Fast Validation
            command.Validate();

            if (command.Invalid)
                return new GenericCommandResult(false, "Erro ao marcar a tarefa como concluída", command.Notifications);

            //Recupera o TodoItem
           **var todo = _repository.GetById(command.Id, command.User);**

             //Altera o título
             todo.MarkAsDone();

            //Salva no banco
            _repository.Update(todo);

            return new GenericCommandResult(true, "Tarefa marcada como concluída com sucesso !", todo);
   }

//Chamando o método **GetById da classe _Queries_**, no Handler:

public ICommandResult Handle(MarkToDoAsDoneCommand command)
        {
            //Fail Fast Validation
            command.Validate();

            if (command.Invalid)
                return new GenericCommandResult(false, "Erro ao marcar a tarefa como concluída", command.Notifications);

            //Recupera o TodoItem
            // var todo = _repository.GetById(command.Id, command.User);
             var todo = Todo.Domain.Queries.TodoQueries.GetById(command.Id);

             //Altera o título
             todo.MarkAsDone();

            //Salva no banco
            _repository.Update(todo);

            return new GenericCommandResult(true, "Tarefa marcada como concluída com sucesso !", todo);
        }

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @rafaelnardomello como vai?

Pode remover do repositório, mas não faz sentido mover o método para as queries... as queries até onde me lembro tem o objetivo apenas de reter as expressions.

Talvez o que você queira seja separar os repositórios em leitura e escrita, correto? Neste caso você pode ter um `ITodoReadRepository` e um `ITodoWritteRepository`, inclusive salvando em bancos diferentes, estando em projetos diferentes.

Era isto?
